### PR TITLE
boards: Add nrf53 bsim

### PR DESCRIPTION
**Pull request by auroraslb**
This PR adds support for simulating the NRF53 network core. To be able to use bluetooth, you will need https://github.com/zephyrproject-rtos/zephyr/pull/30732.

Note: Bluetooth does not fully work, will need some help debugging this.

### CODE REVIEW

File: boards/posix/nrf5x_bsim/irq_handler.c [hunk 1]
```diff
@@ -29,6 +29,42 @@ static struct _isr_list irq_vector_table[NRF_HW_NBR_IRQs];
 
 static int currently_running_irq = -1;
 
+#if defined(CONFIG_SOC_SERIES_NRF53X) || \
+    defined(CONFIG_SOC_COMPATIBLE_NRF53X_CPUNET) || \
+    defined(NRF53_SERIES)
+const char *irqnames[] = { /*just for the traces*/
```
Comment (rugeGerritsen) [inline boards/posix/nrf5x_bsim/irq_handler.c:35]: You may consider adding the remaining IRQ names as well, see nrf5340_network.h. 
* IPC
* EGU

**Review by rugeGerritsen — COMMENTED** [review]
You simplify some conditionals similar to:
`defined(CONFIG_SOC_SERIES_NRF53X) || defined(CONFIG_SOC_COMPATIBLE_NRF53X_CPUNET)`
by adding `select SOC_COMPATIBLE_NRF53X_CPUNET` to `SOC_NRF5340_CPUNET` in`zephyr/soc/arm/nordic_nrf/nrf53/Kconfig.soc`

**Comment by galak** [comment]
Is the intent to replace nrf52_bsim?

**Comment by rugeGerritsen** [comment]
> Is the intent to replace nrf52_bsim?

@galak. I would assume nrf53_bsim can be used as an addition to nrf52_bsim. The HAL is slightly different between nrf52 and nrf53 series devices. @cvinayak and @carlescufi  may comment.

**Comment by galak** [comment]
> @galak. I would assume nrf53_bsim can be used as an addition to nrf52_bsim. The HAL is slightly different between nrf52 and nrf53 series devices. @cvinayak and @carlescufi may comment.

it would be good to explain what the difference / value is between the two.  This is more about a question from a testing/ci point of view.

**Pull request closed without merging**